modules/appeals_api/app/services/appeals_api/uploads/document_validation.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'pdf_info'
+
+module AppealsApi
+  module Uploads
+    class DocumentValidation
+      def initialize(upload)
+        @document = upload[:document]
+      end
+
+      def validate
+        pdf_metadata_present?
+        raise UploadValidationError, file_size_error unless valid_file_size?
```

Comment:
Should we be raising here? The idea is the return meaningful data about the data validation to the client, so they can change/improve on what they're submitting.

An example from the matsumoto RFC documenting the pattern. I don't know if the schema should look like this, but something to think about.

```
   "attachments": [
                        {
                            "page_count": 1,
                            "dimensions": {
                                "height": 8.5,
                                "width": 11.0,
                                "oversized_pdf": false
                            }
                        },
                        {
                            "page_count": 1,
                            "dimensions": {
                                "height": 8.5,
                                "width": 11.0,
                                "oversized_pdf": false
                            }
                        }
                    ]
```

Reply:
This is great info. Thank you for sharing.